File: src/main/java/net/b0n541/pmcts/game/tictactoe/HumanTicTacToePlayer.java
package net.b0n541.pmcts.game.tictactoe;

import java.util.Scanner;

public final class HumanTicTacToePlayer extends TicTacToePlayer {

    public HumanTicTacToePlayer(final PlayerSymbol playerSymbol, final PlayerSymbol firstPlayer) {
        super(playerSymbol, firstPlayer);
    }

    @Override
    public TicTacToeMove play() {
        System.out.println("Your move...");
        System.out.print("Row: ");
        final Scanner scanner = new Scanner(System.in);
        final int row = scanner.nextInt();
        System.out.print("Column: ");
        final int column = scanner.nextInt();

        return new TicTacToeMove(playerSymbol, row, column);
    }
}

File: src/main/kotlin/net/b0n541/ai/game/tictactoe/HumanTicTacToePlayer.kt
package net.b0n541.ai.game.tictactoe

import java.util.*

class HumanTicTacToePlayer(playerSymbol: PlayerSymbol, firstPlayer: PlayerSymbol) :
    TicTacToePlayer(playerSymbol, firstPlayer) {
    override fun play(): TicTacToeMove {
        println("Your move...")
        print("Row: ")
        val scanner = Scanner(System.`in`)
        val row = scanner.nextInt()
        print("Column: ")
        val column = scanner.nextInt()
        return TicTacToeMove(playerSymbol, row, column)
    }
}